core/src/main/scala/kafka/server/Kip500Broker.scala

```diff
@@ -111,7 +111,17 @@ class Kip500Broker(val config: KafkaConfig,
   var quotaManagers: QuotaFactory.QuotaManagers = null
 
   val brokerMetaPropsFile = "meta.properties"
-  val brokerMetadataCheckpoints = config.logDirs.map(logDir => (logDir, new BrokerMetadataCheckpoint(new File(logDir + File.separator + brokerMetaPropsFile)))).toMap
+
+  // Look for metadata checkpoints (meta.properties) in all log.dirs and metadata.log.dir
+  val brokerLogDirs = config.metadataLogDir match {
```

Comment:
Hmm, I still don't think this is quite right.  How about a HashSet or something to make sure there are no duplicates?

Reply:
Yea, I agree. That's cleaner. Pushing a change